sycl/source/handler.cpp

```diff
@@ -278,10 +278,9 @@ event handler::finalize() {
       DiscardEvent = !KernelUsesAssert;
     }
 
-    if (DiscardEvent) {
-      if (PI_SUCCESS != EnqueueKernel())
-        throw runtime_error("Enqueue process failed.",
-                            PI_ERROR_INVALID_OPERATION);
+    if (DiscardEvent && PI_SUCCESS != EnqueueKernel()) {
+      throw runtime_error("Enqueue process failed.",
+                          PI_ERROR_INVALID_OPERATION);
     } else {
```

Comment:
This is a functional change, which we should not make.

Reply:
Thanks! Done.